lib/liblink/cluster/discover.ex

```diff
@@ -0,0 +1,79 @@
+# Copyright (C) 2018  Xerpa
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule Liblink.Cluster.Discover do
+  use Liblink.Cluster.Database.Hook
+
+  alias Liblink.Network.Consul
+  alias Liblink.Data.Consul.Config
+  alias Liblink.Data.Cluster
+  alias Liblink.Data.Cluster.Discover
+  alias Liblink.Cluster.FoldServer
+  alias Liblink.Cluster.ClusterSupervisor
+  alias Liblink.Cluster.Discover.Service
+  alias Liblink.Cluster.Database.Query
+  alias Liblink.Cluster.Database.Mutation
+
+  @impl true
+  def after_hook(pid, tid, event) do
+    case event do
+      {:put, key, _, value} ->
```

Comment:
Se o mesmo _cluster_ for adicionado para ser descoberto duas vezes seguidas, não vai iniciar dois `FoldServers` para a mesma coisa?

Reply:
a chave que insere o cluster usa `put_new`. então adicionar o cluster uma segunda vez não é permitido. é necessário remover e depois adicionar p/ fazer isso.